pkg/identityserver/notification_registry.go

```diff
@@ -165,8 +166,13 @@ func (is *IdentityServer) lookupNotificationReceivers(ctx context.Context, req *
 	return receiverUserIDs, nil
 }
 
+var errNoEmailReceiver = errors.DefineInvalidArgument("no_email_receiver", "no email receiver")
+
 func (is *IdentityServer) storeNotification(ctx context.Context, req *ttnpb.CreateNotificationRequest, receiverUserIDs ...*ttnpb.UserIdentifiers) (*ttnpb.Notification, error) {
 	var notification *ttnpb.Notification
+	if req.Receivers == nil {
```

Comment:
I think it would be better in `is.notifyInternal` to run `req.ValidateFields()` and return the error there than check this here (or even `panic` for that matter).

Also in general when you have slices, use a `len == 0` check instead of a `nil` check. There are non-`nil` slices with length 0.